From d5541ab9b0ad182611cedc91d87bb935cbde0d2a Mon Sep 17 00:00:00 2001
From: Leonard Crestez <leonard.crestez@nxp.com>
Date: Wed, 12 Sep 2018 16:00:50 +0300
Subject: [PATCH 4619/5242] MLK-19368: Revert "MLK-17921-3 driver: irqchip:
 shorten wakeup wait time of IPI SW workaround"

commit  9871cee33749d5a37b59ae0cdf9fb3de182a3e31 from
https://source.codeaurora.org/external/imx/linux-imx.git

This reverts commit 1d965a3e12797e6561df8b5fcc16ad1d60182dab.

Shrinking the udelay from 50 to 10 causes lost IPIs and hangs on
suspend so let's revert to behavior from previous releases.

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Acked-by: Jun Li <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/irqchip/irq-gic-v3.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/irqchip/irq-gic-v3.c b/drivers/irqchip/irq-gic-v3.c
index 7130fda..3ec9eb9 100644
--- a/drivers/irqchip/irq-gic-v3.c
+++ b/drivers/irqchip/irq-gic-v3.c
@@ -767,7 +767,7 @@ static void gic_raise_softirq(const struct cpumask *mask, unsigned int irq)
 			val |= (1 << 12);
 			writel_relaxed(val, iomuxc_gpr_base + 0x4);
 			/* delay for a while to make sure cores wakeup done */
-			udelay(10);
+			udelay(50);
 			val &= ~(1 << 12);
 			writel_relaxed(val, iomuxc_gpr_base + 0x4);
 		}
-- 
1.7.9.5

